general/raja.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+#ifndef MFEM_RAJA_HPP
+#define MFEM_RAJA_HPP
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_RAJA
+#include "RAJA/RAJA.hpp"
+#endif
+
+#if defined(MFEM_USE_RAJA) && defined(RAJA_ENABLE_CUDA)
+
+namespace RAJA
+{
+
+namespace cuda
+{
+
+__device__ __forceinline__ int globalIdx2D()
+{
+   return blockIdx.x * blockDim.z + threadIdx.z;
+}
+
+__device__ __forceinline__ int globalIdx3D() { return blockIdx.x; }
+
+template <bool batch, typename Iterator, typename DBODY, typename IndexType>
+__global__ void RajaKernelND(DBODY body, Iterator I, IndexType N)
+{
+   const auto k = static_cast<IndexType>(batch ? globalIdx2D():globalIdx3D());
+   if (k >= N) { return; }
+   body(I[k]);
+}
```

Comment:
I have an idea.... try replacing lines 38-40 with 
 ```
const int k = blockIdx.x;
  if (k >= N) { return; }
  body(k); 
```
like you have in your CUDA code and try a 3D problem. Just for benchmarking purposes....

Reply:
Tried, but it does not really change the performances.